File: DSA/Cpp/test/dsa/lib/math/nums/max_sliding_window_test.cpp
#include <dsa/lib/math/nums/max_sliding_window.hpp>
#include "nums_test.h"
namespace dsa::lib::math::nums {
namespace {

class MaxSlidingWindowTest : public NumsTest {
 protected:
    static std::vector<int> maxSlidingWindow(std::vector<int> &nums, int k) {
        return MaxSlidingWindow::maxSlidingWindow(nums, k);
    }
};
TEST_F(MaxSlidingWindowTest, case1) {
    std::vector<int> nums = {1, 3, -1, -3, 5, 3, 6, 7};
    std::vector<int> expected = {3, 3, 5, 5, 6, 7};
    std::vector<int> actual = maxSlidingWindow(nums, 3);
    EXPECT_EQ(expected, actual);
}
TEST_F(MaxSlidingWindowTest, case2) {
    std::vector<int> nums = {1};
    std::vector<int> expected = {1};
    std::vector<int> actual = maxSlidingWindow(nums, 1);
    EXPECT_EQ(expected, actual);
}
TEST_F(MaxSlidingWindowTest, case3) {
    std::vector<int> nums = {1, -1};
    std::vector<int> expected = {1, -1};
    std::vector<int> actual = maxSlidingWindow(nums, 1);
    EXPECT_EQ(expected, actual);
}
TEST_F(MaxSlidingWindowTest, case4) {
    std::vector<int> nums = {9, 11};
    std::vector<int> expected = {11};
    std::vector<int> actual = maxSlidingWindow(nums, 2);
    EXPECT_EQ(expected, actual);
}
TEST_F(MaxSlidingWindowTest, case5) {
    std::vector<int> nums = {4, -1};
    std::vector<int> expected = {4};
    std::vector<int> actual = maxSlidingWindow(nums, 2);
    EXPECT_EQ(expected, actual);
}

}
}

File: DSA/Java/test/dsa/lib/math/nums/MaxSlidingWindowTest.java
package dsa.lib.math.nums;

import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class MaxSlidingWindowTest {
    private int[] maxSlidingWindow(int[] nums, int k) {
        return new MaxSlidingWindow().maxSlidingWindow(nums, k);
    }

    @Test
    public void case1() {
        assertArrayEquals(new int[]{4}, maxSlidingWindow(new int[]{4, -1}, 2));
        assertArrayEquals(new int[]{11}, maxSlidingWindow(new int[]{9, 11}, 2));
        assertArrayEquals(new int[]{1, -1}, maxSlidingWindow(new int[]{1, -1}, 1));
        assertArrayEquals(new int[]{1}, maxSlidingWindow(new int[]{1}, 1));
        assertArrayEquals(new int[]{3, 3, 5, 5, 6, 7}, maxSlidingWindow(new int[]{1, 3, -1, -3, 5, 3, 6, 7}, 3));
    }
}